### PR TITLE
chore: update and clean dependencies

### PR DESCRIPTION
**Pull request by mcasimir**
Add depcheck and fix dependencies: 
- remove unused deps
- add missing deps
- move packages not used in prod to devDependencies
- update mocha and chai to recent versions
- updates `bson` to the version deduped in compass

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,18 +33,17 @@
...
   "dependencies": {
     "antlr4": "4.7.2",
-    "bson": "^4.0.2",
-    "chai": "^4.1.2",
-    "context-eval": "^0.1.0",
-    "fast-json-parse": "^1.0.3",
-    "js-yaml": "^3.11.0",
-    "mocha": "^5.0.2",
-    "mongodb": "^3.3.0"
+    "bson": "4.1.0",
```
Comment (addaleax) [inline package.json:46]: Wouldn’t we want the deduped version in Compass to be 4.4.0?

**Review by addaleax — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (addaleax) [reply package.json:46]: (I’ll just go try to do that for COMPASS-4502)

Comment (addaleax) [reply package.json:46]: Okay, that’s still not happening because of https://jira.mongodb.org/browse/NODE-3376.

**Comment by addaleax** [comment]
Also, random fun fact:
Screenshot from 2021-06-22 17-26-30

**Comment by mcasimir** [comment]
> Also, random fun fact:
> Screenshot from 2021-06-22 17-26-30

Yeah forgot to install `depcheck`

**Comment by addaleax** [comment]
@mcasimir I was more pointing towards the fact that e.g. I don’t have merge access here

**Pull request merged**